### PR TITLE
[MINOR][DOCS] Add zstandard as a candidate to fix the desc of spark.sql.avro.compression.codec

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

Add zstandard as a candidate to fix the desc of spark.sql.avro.compression.codec

### Why are the changes needed?

docfix

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

doc build
### Was this patch authored or co-authored using generative AI tooling?

no

### CODE REVIEW

File: docs/sql-data-sources-avro.md [hunk 1]
```diff
@@ -348,7 +348,7 @@ Configuration of Avro can be done via `spark.conf.set` or by running `SET key=va
     <td>snappy</td>
     <td>
       Compression codec used in writing of AVRO files. Supported codecs: uncompressed, deflate,
-      snappy, bzip2 and xz. Default codec is snappy.
+      snappy, bzip2, xz and zstandard. Default codec is snappy.
```
Comment (nchammas) [inline docs/sql-data-sources-avro.md:351]: I suggest removing this bit as it duplicates the information in the default column.

```suggestion
      snappy, bzip2, xz and zstandard.
```

Same here, of course: https://github.com/apache/spark/blob/bbeb8d7417bafa09ad5202347175a47b3217e27f/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L3604

**Review by nchammas — APPROVED** [review]
If I may advertise my own work for a moment, adopting either the approach in #44755 or #44756 will solve this problem of the HTML config docs getting out of sync with the source code. 😊

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @yaooqinn .

**Comment by dongjoon-hyun** [comment]
Since it's supported since Spark 3.2.0 via SPARK-34479, I backported this to all live release branches. Merged to main/3.5/3.4.

**Comment by yaooqinn** [comment]
Thank @dongjoon-hyun @nchammas

**Pull request closed without merging**